### PR TITLE
toolbar refresh

### PR DESCRIPTION
**Pull request by eoji**
## what
Started with discovery as a first pass since it's most visible. `AppBarLayout` no longer has elevation, now has a divider whose alpha is based on how much the user has scrolled. The toolbar is only expanded when at the top of the list.

For fun, I got rid of the awkward selected state in the toolbar on pre 21 devices.

### CODE REVIEW

File: app/src/main/java/com/kickstarter/ui/toolbars/AlphaBehavior.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import android.util.AttributeSet;
+import android.view.View;
+
+
+public class AlphaBehavior<V extends View> extends CoordinatorLayout.Behavior<V> {
+
+  public AlphaBehavior() {
+  }
+
+  public AlphaBehavior(final Context context, final AttributeSet attrs) {
```
Comment (luoser) [inline app/src/main/java/com/kickstarter/ui/toolbars/AlphaBehavior.java]: nice use of a custom view class!!!!!!! only comment here are to annotate the method params `@NonNull`

File: app/src/main/res/drawable/click_indicator_light.xml [hunk 2]
```diff
@@ -2,6 +2,5 @@
 <selector xmlns:android="http://schemas.android.com/apk/res/android">
   <item android:drawable="@color/dark_gray_click" android:state_focused="true" />
   <item android:drawable="@color/dark_gray_click" android:state_pressed="true" />
-  <item android:drawable="@color/dark_gray_click" android:state_selected="true" />
```
Comment (luoser) [inline app/src/main/res/drawable/click_indicator_light.xml]: aha this was showing the awkward selected state? thanks!

File: app/src/main/res/layout/discovery_layout.xml [hunk 3]
```diff
@@ -73,6 +60,12 @@
         android:layout_width="match_parent"
         android:layout_height="match_parent"/>
 
+      <include
+        layout="@layout/horizontal_line_1dp_view"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:layout_behavior="com.kickstarter.ui.toolbars.AlphaBehavior" />
```
Comment (luoser) [inline app/src/main/res/layout/discovery_layout.xml]: sweeeeeeeeet 🍭

File: app/src/main/res/layout/discovery_layout.xml [hunk 4]
```diff
@@ -11,60 +11,47 @@
   tools:context=".ui.activities.DiscoveryActivity">
 
   <RelativeLayout
-    android:layout_marginTop="@dimen/discovery_drawer_status_bar_height_negative"
-    android:paddingTop="@dimen/discovery_drawer_status_bar_height"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:fitsSystemWindows="true">
 
-    <android.support.design.widget.AppBarLayout
```
Comment (luoser) [inline app/src/main/res/layout/discovery_layout.xml]: thanks for cleaning up this unneeded AppBarLayout!

**Comment by eoji** [comment]
[More refresh](https://trello.com/c/Bb0vAdOx/293-more-refresh)

**Pull request merged**